jsonschema/_typing.py

```diff
@@ -0,0 +1,14 @@
+"""
+Type aliases and utilities to help manage `typing` usage and type annotations.
+"""
+
+from collections.abc import Mapping, Sequence
+import typing
+
+Schema = Mapping[str, typing.Any]
```

Comment:
Schema is tricky to nail down for multiple reasons. Its precise type depends on the specific dialect of JSON Schema involved, and (speaking off the top of my head without double checking, which counts for lots of the comments I'm about to make --)
`applicable_validators` means that *technically* its type is literally `Any` (or some dependent type), because someone can decide they're inventing a JSON Schema dialect where the number `17` is a schema and means something, and `applicable_validators` will give the library what it needs to deal with that as a schema. I'm honestly not sure what to do about that.
* Ignoring the above, the "internal" type to all JSON Schema "official" drafts is -- for later drafts, `bool | Mapping[str, typing.Any]`, and for earlier ones `Mapping[str, typing.Any]`

Reply:
I'm definitely going to have to circle back to try to better grasp what's sensible here.

In line with my other comment about maybe accepting some technical inaccuracy in the annotations as a pragmatic approach, I'm instinctively inclined towards `bool | Mapping[...]`.